### PR TITLE
Per Frame Handler Code Improvement

### PR DESCRIPTION
**Pull request by jokoho48**
This PR include:
Updated Parameterization in some Functions.
3 new Function for find the First Entry in a Array that is Nil, Null or TypeOf.
Fix a Issue where PFH get Stacked
Improve Performance in PFH Code
A Updated RemovePFH Comment/Fix Issues there

### CODE REVIEW

File: addons/arrays/CfgFunctions.hpp [hunk 1]
```diff
@@ -15,6 +15,24 @@ class CfgFunctions
...
+                file = "\x\cba\addons\arrays\fnc_findFirstEmptyEntry.sqf";
+            };
+            // CBA_fnc_findFirstTypeOfEntry
+            class findFirstTypeOfEntry
+            {
+                description = "A function that return the index of the first Type Of Entry in a Array.";
+                file = "\x\cba\addons\arrays\fnc_findFirstTypeOfEntry.sqf";
+            };
+            // CBA_fnc_findFirstTypeOfEntry
+            class CBA_fnc_findFirstNullEntry
```
Comment (jonpas) [inline addons/arrays/CfgFunctions.hpp]: Bad copy/paste. `CBA_fnc_` should not be here.

File: addons/arrays/fnc_findFirstEmptyEntry.sqf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    joko // Jonas
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+scopeName "main";
+
+{
+    private "_current";
+    _current = _x;
+    if (isNil "_current") then {
```
Comment (Glowbal) [inline addons/arrays/fnc_findFirstEmptyEntry.sqf]: Won't `isNil "_x"` work here?

File: addons/common/fnc_removePerFrameHandler.sqf [hunk 3]
```diff
@@ -25,19 +25,20 @@ Author:
...
-        _newArray set[_i, _entry];
+
+{
+    private "_entry";
+    _entry = _x;
+    if !(isNil "_entry") then {
+        _newArray set [_forEachIndex, _entry];
     };
-};
+} forEach GVAR(perFrameHandlerArray);
```
Comment (Glowbal) [inline addons/common/fnc_removePerFrameHandler.sqf]: Wouldn't this break any stored PFH Ids?

Comment (jokoho48) [inline addons/common/fnc_removePerFrameHandler.sqf]: no

File: addons/arrays/fnc_findFirstEmptyEntry.sqf (hunk 2 above)
Comment (jokoho48) [inline addons/arrays/fnc_findFirstEmptyEntry.sqf]: i dont know not tested

**Comment by jokoho48** [comment]
ok i found a better Method for PFH and i will split this in to 2 PRs one for the 3 New Function (that are not more needed for PFH) and one for the PFH Performance Improvements

**Pull request closed without merging**